src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -161,6 +161,18 @@ private static void AddCollectionRuleActionSchema<TOptions>(GenerationContext co
 
             settingsProperty.Reference = context.AddTypeIfNotExist<TOptions>();
 
+            // Don't require properties that have a corresponding collection rule default
+            foreach (var propName in typeof(CollectionRuleDefaultsOptions).GetProperties().Select(x => x.Name))
+            {
+                settingsProperty.Reference.RequiredProperties.Remove(propName);
+            }
+
+            // Have to special case for Traces since Profile/Providers aren't listed as required, but one of them must be included
+            if (settingsProperty.Reference.RequiredProperties.Count > 0 || typeof(TOptions) == typeof(CollectTraceOptions))
```

Comment:
Let's not special case this particular set of options. I don't want us to have to add more special cases for future actions. The algorithm should say that "if a property is required, does not have a default value, and has no corresponding default property, then the settings are required".